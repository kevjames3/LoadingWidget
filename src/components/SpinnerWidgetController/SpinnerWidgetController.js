import React, { Component } from "react";
import "./styles/main.css";

class SpinnerWidgetController extends Component {
  render() {
    return (
      <div className="spinner-widget-controls">
        <div className="button-controls">
          <button
            className="start-spinning"
            onClick={() => this.props.onSpinningChange(true)}
          >
            Keep Spinning
          </button>
          <button
            className="stop-spinning"
            onClick={() => this.props.onSpinningChange(false)}
          >
            Stop Spinning
          </button>
        </div>
        <div className="percentage-input">
          <label>Current percentage</label>
          <input
            type="Number"
            defaultValue={this.props.value === undefined ? 0 : this.props.value}
            onInput={(event) => {
              this.props.onValueChange(event.target.value);
            }}
            min="0"
            max="100"
          ></input>
        </div>
      </div>
    );
  }
}

export default SpinnerWidgetController;
