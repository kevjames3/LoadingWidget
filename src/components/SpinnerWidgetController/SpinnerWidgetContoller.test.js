import React from "react";
import { render, getByLabelText } from "@testing-library/react";
import renderer from "react-test-renderer";
import SpinnerWidgetController from "./SpinnerWidgetController";

describe("SpinnerWidgetContoller", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SpinnerWidgetController />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("handles the default value correctly", () => {
    const { container } = render(<SpinnerWidgetController />);
    expect(container.querySelector(".percentage-input input")).toHaveValue(0);
  });

  it("handles the different starting values", () => {
    const { container } = render(<SpinnerWidgetController value={25} />);
    expect(container.querySelector(".percentage-input input")).toHaveValue(25);
  });
});
