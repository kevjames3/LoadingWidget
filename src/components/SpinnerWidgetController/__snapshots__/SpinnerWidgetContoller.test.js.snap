// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerWidgetContoller renders correctly 1`] = `
<div
  className="spinner-widget-controls"
>
  <div
    className="button-controls"
  >
    <button
      className="start-spinning"
      onClick={[Function]}
    >
      Keep Spinning
    </button>
    <button
      className="stop-spinning"
      onClick={[Function]}
    >
      Stop Spinning
    </button>
  </div>
  <div
    className="percentage-input"
  >
    <label>
      Current percentage
    </label>
    <input
      defaultValue={0}
      max="100"
      min="0"
      onInput={[Function]}
      type="Number"
    />
  </div>
</div>
`;
